Extensions/timestamps.js

```diff
@@ -224,12 +245,91 @@ XKit.extensions.timestamps = new Object({
 
 			var timestamp = responseData.posts[0].timestamp;
 			date_element.html(this.format_date(moment(new Date(timestamp * 1000))));
-			date_element.removeClass("xtimestamp_loading");
+			date_element.removeClass("xtimestamp-loading");
 			XKit.storage.set("timestamps", "xkit_timestamp_cache_" + post_id, timestamp);
 		})
 		.catch(() => this.show_failed(date_element));
 	},
 
+	react_add_timestamps: function() {
+		var posts = $("[data-id]:not(.xkit_timestamps)");
+		posts.addClass("xkit_timestamps");
+
+		if (posts.length === 0) {
+			return;
+		}
+
+		XKit.extensions.timestamps.check_quota();
```

Comment:
the cache is (or should be) unused for post timestamps, but should be used for reblog timestamps since they have to be fetched; the quota doesn't need to be checked here

```suggestion
```